source/errors.ts

```diff
@@ -1,11 +1,16 @@
-'use strict';
-const urlLib = require('url');
-const http = require('http');
-const PCancelable = require('p-cancelable');
-const is = require('@sindresorhus/is');
-
-class GotError extends Error {
-	constructor(message, error, options) {
+import urlLib from 'url';
+import http, {IncomingHttpHeaders} from 'http';
+import PCancelable from 'p-cancelable';
+import is from '@sindresorhus/is';
+import {Response, Timings, Options} from './utils/types';
+import {TimeoutError as TimedOutError} from './utils/timed-out';
+
+type ErrorWithCode = (Error & { code?: string }) | { code?: string };
```

Comment:
This is used to solve type errors for these cases:

- An `Error` object is passed, which doesn't contain a `code` property.
- An `Error` object is passed, which contains a `code` property. — See [`test/retry.js:165`](https://github.com/sindresorhus/got/blob/master/test/retry.js#L165)
- An object is passed which doesn't contain a `code` property.
- An object is passed which contains a `code` property. — See [`source/errors.js:102`](https://github.com/sindresorhus/got/blob/master/source/errors.js#L102)

Reply:
Thank you for teaching me that!